ReactNativeClient/lib/services/KeymapService.ts

```diff
@@ -0,0 +1,284 @@
+			throw new Error(
+				`"${item.command}" is not a valid command.\n\n` +
+				`Visit ${commandReference} to see the list of available commands.`
+			);
+		}
+
+		if (!item.hasOwnProperty('accelerator')) {
+			throw new Error(
+				'Couldn\'t find the "accelerator" property.\n\n' +
+				`Visit ${acceleratorReference} to learn more about Electron Accelerators.`
+			);
+		} else if (!(item.accelerator === null || KeymapService.isAccelerator(item.accelerator))) {
+			throw new Error(
+				`"${item.accelerator}" is not a valid Accelerator.\n\n` +
+				`Visit ${acceleratorReference} to learn more about Electron Accelerators.`
+			);
+		}
+	}
+
+	private validateKeymap() {
+		const usedAccelerators = new Set();
+
+		for (const item of Object.values(this.keymap)) {
+			const itemAccelerator = item.accelerator;
+
+			if (usedAccelerators.has(itemAccelerator)) {
+				const originalItem = Object.values(this.keymap).find(_item => _item.accelerator == item.accelerator);
+
+				throw new Error(
+					`Accelerator "${itemAccelerator}" can't be used for "${item.command}" command ` +
+					`because it's already used for "${originalItem.command}" command!\n\n` +
+					'You have to change the Accelerator for any of above commands.'
+				);
+			} else if (itemAccelerator !== null) {
+				usedAccelerators.add(itemAccelerator);
+			}
+		}
+	}
+
+	static initKeymap() {
+		const itemsByCommand: { [command: string]: KeymapItem } = {};
+		for (let i = 0; i < KeymapService.defaultKeymap.length; i++) {
+			// Make a copy of the KeymapItem before assigning it, hence the spread operator
+			// Otherwise we're going to mess up the defaultKeymap array
+			itemsByCommand[KeymapService.defaultKeymap[i].command] = { ...KeymapService.defaultKeymap[i] };
+		}
+
+		return itemsByCommand;
+	}
+
+	static isAccelerator(accelerator: string) {
+		let keyFound = false;
+		const parts = accelerator.split('+');
+
+		return parts.every((part, index) => {
+			const isKey = KeymapService.keyCodes.test(part);
+			const isModifier = KeymapService.modifiers.test(part);
+
+			if (isKey) {
+				// Key must be unique
+				if (keyFound) return false;
+				keyFound = true;
+			}
+
+			// Key is required
+			if (index === (parts.length - 1) && !keyFound) return false;
+			return isKey || isModifier;
+		});
+	}
+
+	static localizeAccelerator(accelerator: string) {
+		return accelerator === null
+			? null
+			: accelerator
+				// Allow usage of all modifiers for the sake of compatibility
+				// But only use one type of modifiers to improve duplicate detection
+				.replace(/(CommandOrControl|CmdOrCtrl)/i, shim.isMac() ? 'Cmd' : 'Ctrl')
+				.replace(/(Command)/i, 'Cmd')
+				.replace(/(Control)/i, 'Ctrl')
+				.replace(/(Alt|Option)/i, shim.isMac() ? 'Option' : 'Alt')
+				// Reorder components so that "Option+Shift+Cmd+X" and "Shift+Option+Cmd+X" will be the same
+				.split('+')
+				.sort((a, b) => KeymapService.getWeight(a) - KeymapService.getWeight(b))
+				.join('+');
+	}
```

Comment:
We should be strict about the input we accept, so this function should not be needed, and the getWeight function either. Now that you've split config between macOS and the rest, this is even less necessary since we don't have CommandOrControl anymore.

Reply:
Okay, I'll remove this.